### PR TITLE
TRUNK-5652:Add NestedOrderGroups to OrderGroup

### PR DESCRIPTION
**Pull request by gitcliff**
Llink:https://issues.openmrs.org/browse/TRUNK-5652?src=confmacro

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you also update the changeset count in DatabaseUpdaterDatabaseIT?

**Comment by gitcliff** [comment]
@dkayiwa I have updated the count

**Comment by dkayiwa** [comment]
The changes that need to happen are in the ticket description.

**Comment by gitcliff** [comment]
@dkayiwa   have added some changes ,,,kindly review

**Comment by dkayiwa** [comment]
@gitcliff i noticed that you have added more changes than required as per the ticket description. Can you look into that?

**Comment by gitcliff** [comment]
@dkayiwa i have removed the un-required changes

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml [hunk 1]
```diff
@@ -112,6 +112,25 @@
 								 referencedTableName="order_group"
 								 referencedColumnNames="order_group_id" />
     </changeSet>
+        
+    <changeSet id="20190816-Trunk-5652" author="gitacliff">
+		<preConditions onFail="MARK_RAN">
+			<not>
```
Comment (dkayiwa) [inline api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml:118]: Was this also in the ticket description?

Comment (gitcliff) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml:118]: @dkayiwa  the Preconditions element can be attached to  a changeset to control the execution of an update based on the state of the database and in this case its housing the `<not>` element just like in the all other changesets adding new columns to the order_group table

Comment (dkayiwa) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml:118]: I mean the entire changeset.

Comment (gitcliff) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml]: @dkayiwa i have made some changes ,,please review

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml [hunk 2]
```diff
@@ -112,7 +112,7 @@
 								 referencedTableName="order_group"
 								 referencedColumnNames="order_group_id" />
     </changeSet>
-
```
Comment (dkayiwa) [inline api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml:115]: Do you need to touch this file in any way?

Comment (gitcliff) [reply api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml]: i have fixed it,,thanks

File: api/src/test/java/org/openmrs/util/DatabaseUpdaterDatabaseIT.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
 	 * 
 	 * This constant needs to be updated when adding new Liquibase update files to openmrs-core.
 	 */
-	private static final int CHANGE_SET_COUNT_FOR_2_1_X = 863;
+	private static final int CHANGE_SET_COUNT_FOR_2_1_X = 864;
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/util/DatabaseUpdaterDatabaseIT.java:32]: What is in core is already 864. Can you increment this again?

Comment (gitcliff) [reply api/src/test/java/org/openmrs/util/DatabaseUpdaterDatabaseIT.java]: @dkayiwa  I have incremented

File: api/src/main/resources/org/openmrs/api/db/hibernate/OrderGroup.hbm.xml [hunk 4]
```diff
@@ -72,6 +72,12 @@
 		    not-null="false" column="parent_order_group" />
 		<many-to-one name="orderGroupReason" class="org.openmrs.Concept"
 		    not-null="false" column="order_group_reason" />
+		 <!-- one-to-many association to all nested orderGroups in this grouping -->
+        <set name="nestedOrderGroups" inverse="true" cascade="delete" order-by="order_group_id"
+            access="field" batch-size="25">
+            <key column="order_group_id" />
```
Comment (dkayiwa) [inline api/src/main/resources/org/openmrs/api/db/hibernate/OrderGroup.hbm.xml:78]: This does not look like that in the commit that i linked in the ticket description.

Comment (gitcliff) [reply api/src/main/resources/org/openmrs/api/db/hibernate/OrderGroup.hbm.xml]: i have fixed it

**Comment by dkayiwa** [comment]
Have you implemented all that is in the ticket description?

**Comment by gitcliff** [comment]
@dkayiwa  i have added some changes ,,,

**Pull request merged**